.github/workflows/ci.yml

```diff
@@ -39,16 +78,24 @@ jobs:
         name: 'Checkout'
         with:
           fetch-depth: 0
+          # If the workflow wasn't triggered by dispatch, this will be empty and use defaults.
+          ref: ${{ needs.dispatch-handler.outputs.head }}
+
       - uses: './.github/actions/setup-woocommerce-monorepo'
         name: 'Setup Monorepo'
         with:
           php-version: false # We don't want to waste time installing PHP since we aren't using it in this job.
+
       - uses: actions/github-script@v7
         name: 'Build Matrix'
         id: 'project-jobs'
+        env:
+          PR_SIM: ${{ needs.dispatch-handler.outputs.base }}
         with:
           script: |
-            let baseRef = ${{ toJson( github.base_ref ) }};
+            const prSim = process.env.PR_SIM;
+            
+            let baseRef = prSim || ${{ toJson( github.base_ref ) }};
```

Comment:
Out of curiosity, why don't we wrap `prSim` into `toJson`?

```
    let baseRef = ${{ toJson( needs.dispatch-handler.outputs.base || github.base_ref ) }};
```

Reply:
I guess that should work as well and looks more elegant. It's Jonathan's initial implementation, so I cannot answer on his thinking. 
I would rather not make such changes at this point, that will require another round of time consuming testing in a fork.

Reply:
Gotcha! If it works, we are ok to keep it as it is.